language-plutus-core/test/Spec.hs

```diff
@@ -188,18 +188,26 @@ testRebindShadowedVariable =
     let
         xName = TyName (Name () "x" (Unique 0))
         yName = TyName (Name () "y" (Unique 1))
+        zName = TyName (Name () "z" (Unique 2))
 
         varX = TyVar () xName
         varY = TyVar () yName
+        varZ = TyVar () zName
 
         typeKind = Type ()
 
         -- (all x (type) (fun (all y (type) y) x))
-        type0 = TyForall () xName typeKind (TyFun () (TyForall () yName typeKind varY) varX)
+        l1 = TyForall () xName typeKind (TyFun () (TyForall () yName typeKind varY) varX)
         -- (all x (type) (fun (all x (type) x) x))
-        type1 = TyForall () xName typeKind (TyFun () (TyForall () xName typeKind varX) varX)
+        r1 = TyForall () xName typeKind (TyFun () (TyForall () xName typeKind varX) varX)
+
+        -- (all x (type) (all x (type) (fun x x)))
+        l2 = TyForall () xName typeKind (TyForall () xName typeKind (TyFun () varX varX))
+        -- (all y (type) (all z (type) (fun y z)))
+        r2 = TyForall () yName typeKind (TyForall () zName typeKind (TyFun () varY varZ))
+
     in
-        type0 == type1
+        l1 == r1 && l2 == r2 {- THE LATTER MUST NOT PASS -}
```

Comment:
Unrelated changes in here?

Reply:
Oh, I changed the branch, but I guess did something wrong.